lightning/src/ln/channelmanager.rs

```diff
@@ -18295,7 +18348,7 @@ where
 			pending_intercepted_htlcs: Mutex::new(pending_intercepted_htlcs_legacy),
 
 			forward_htlcs: Mutex::new(forward_htlcs_legacy),
-			decode_update_add_htlcs: Mutex::new(decode_update_add_htlcs_legacy),
+			decode_update_add_htlcs: Mutex::new(decode_update_add_htlcs),
```

Comment:
I looked at the final commit of this PR, and there it seems that `decode_update_add_htlcs_legacy` isn't used at all anymore. Shouldn't it still be input in case we don't have data from monitor(s)? Maybe I am overlooking something.

Reply:
Good catch, I fixed this and added a test.